tests/unattended/unit/framework/bach.sh

```diff
@@ -0,0 +1,642 @@
+# -*- mode: sh -*-
+# Bach Testing Framework, https://bach.sh
+# Copyright (C) 2019  Chai Feng <chaifeng@chaifeng.com>
+#
+# Bach Testing Framework is dual licensed under:
+# - GNU General Public License v3.0
+# - Mozilla Public License 2.0
+set -euo pipefail
+shopt -s expand_aliases
+
+export BACH_COLOR="${BACH_COLOR:-auto}"
+export PS4='+(${BASH_SOURCE##*/}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+
+BACH_OS_NAME="$(uname)"
+declare -gxr BACH_OS_NAME
+
+declare -gxa bach_origin_paths=()
+while builtin read -r -d: folder; do
+    bach_origin_paths+=("$folder")
+done <<< "${PATH}:"
+
+function @out() {
+    if [[ "${1:-}" == "-" || ! -t 0 ]]; then
+        [[ "${1:-}" == "-" ]] && shift
+        while IFS=$'\n' read -r line; do
+            builtin printf "%s\n" "${*}$line"
```

Comment:
Please use ${var} notation to reference variable values

Reply:
`bach.sh` is the external framework. We will not modify it.